pkg/scheduler/server/api.go

```diff
@@ -61,111 +66,93 @@ func (s *Server) ScheduleJob(ctx context.Context, req *schedulerv1pb.ScheduleJob
-func (s *Server) ListJobs(ctx context.Context, req *schedulerv1pb.ListJobsRequest) (*schedulerv1pb.ListJobsResponse, error) {
-	select {
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	case <-s.readyCh:
-	}
+		invokeMethod := "remind/" + reminderName
+		contentType := metadata["content-type"]
+		invokeReq := internalv1pb.NewInternalInvokeRequest(invokeMethod).
+			WithActor(actorType, actorID).
+			WithData(payload.GetValue()).
+			WithContentType(contentType)
 
-	entries := s.cron.ListJobsByPrefix(req.GetAppId() + "||")
-
-	jobs := make([]*runtimev1pb.Job, 0, len(entries))
-	for _, entry := range entries {
-		job := &runtimev1pb.Job{
-			Name:     entry.Name,
-			Schedule: entry.Rhythm,
-			Repeats:  entry.Repeats,
-			DueTime:  entry.DueTime,
-			Ttl:      entry.TTL,
-			Data:     entry.Data,
+		res, err := s.actorRuntime.Call(ctx, invokeReq)
+		if err != nil {
+			return etcdcron.Failure, err
 		}
 
-		jobs = append(jobs, job)
-	}
+		if res.GetStatus().GetCode() != int32(codes.OK) {
+			return etcdcron.Failure, nil
+		}
 
-	resp := &schedulerv1pb.ListJobsResponse{Jobs: jobs}
+		return etcdcron.OK, err
+	}
 
-	return resp, nil
+	// Trigger not supported
+	log.Warn("Cannot trigger job: not supported.")
+	return etcdcron.Failure, nil
 }
 
-// GetJob is a placeholder method that needs to be implemented
-func (s *Server) GetJob(ctx context.Context, req *schedulerv1pb.JobRequest) (*schedulerv1pb.GetJobResponse, error) {
+func (s *Server) DeleteJob(ctx context.Context, req *schedulerv1pb.JobRequest) (*schedulerv1pb.DeleteJobResponse, error) {
 	select {
 	case <-ctx.Done():
 		return nil, ctx.Err()
 	case <-s.readyCh:
 	}
 
-	job := s.cron.GetJob(req.GetJobName())
-	if job != nil {
-		resp := &schedulerv1pb.GetJobResponse{
-			Job: &runtimev1pb.Job{
-				Name:     job.Name,
-				Schedule: job.Rhythm,
-				Repeats:  job.Repeats,
-				DueTime:  job.DueTime,
-				Ttl:      job.TTL,
-				Data:     job.Data,
-			},
-		}
-		return resp, nil
+	jobName := composeJobName(req.GetNamespace(), req.GetJobName())
+	err := s.cron.DeleteJob(ctx, jobName)
+	if err != nil {
+		log.Errorf("error deleting job %s: %s", jobName, err)
+		return nil, err
 	}
 
-	return nil, fmt.Errorf("job not found")
+	return &schedulerv1pb.DeleteJobResponse{}, nil
 }
 
-// DeleteJob is a placeholder method that needs to be implemented
-func (s *Server) DeleteJob(ctx context.Context, req *schedulerv1pb.JobRequest) (*schedulerv1pb.DeleteJobResponse, error) {
-	select {
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	case <-s.readyCh:
+// parseStartTime is a wrapper around timeutils.ParseTime that truncates the time to seconds.
+func parseStartTime(dueTime string) (time.Time, error) {
+	if dueTime == "" {
+		return time.Time{}, nil
 	}
 
-	err := s.cron.DeleteJob(req.GetJobName())
+	now := time.Now()
+	t, err := timeutils.ParseTime(dueTime, &now)
 	if err != nil {
-		log.Errorf("error deleting job %s: %s", req.GetJobName(), err)
-		return nil, err
+		return t, err
```

Comment:
```suggestion
		return nil, err
```

Reply:
nil is not a valid return for time.Time.